probes/contributorsFromOrgOrCompany/impl_test.go

```diff
@@ -0,0 +1,171 @@
+	Companies        []string
+	Organizations    []User
+	NumContributions int
+	ID               int64
+	IsBot            bool
+}
+
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	// nolint:govet
+	tests := []struct {
+		name     string
+		raw      *checker.RawResults
+		outcomes []finding.Outcome
+		err      error
+	}{
+		{
+			name: "Should be 'comp1', 'comp2', 'Login'",
+			raw: &checker.RawResults{
+				ContributorsResults: checker.ContributorsData{
+					Users: []clients.User{
+						{
+							Companies: []string{"comp1", "comp2"},
+							Organizations: []clients.User{
+								{
+									Login:            "Login",
+									Companies:        []string{"comp3", "comp4"}, // These should not be included
+									NumContributions: 10,
+								},
+							},
+							NumContributions: 10,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+			},
+		}, {
+			name: "Should be 'comp1', 'comp2', 'comp3', 'comp4', 'Login1', 'Login2'",
+			raw: &checker.RawResults{
+				ContributorsResults: checker.ContributorsData{
+					Users: []clients.User{
+						{
+							Companies: []string{"comp1", "comp2"},
+							Organizations: []clients.User{
+								{
+									Login:            "Login1",
+									NumContributions: 10,
+								},
+							},
+							NumContributions: 10,
+						},
+						{
+							Companies: []string{"comp3", "comp4"},
+							Organizations: []clients.User{
+								{
+									Login:            "Login2",
+									NumContributions: 10,
+								},
+							},
+							NumContributions: 10,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+			},
+		}, {
+			name: "Should be 'comp1', 'comp2', 'Login1' V2. The second user has too few contributions to be considered",
+			raw: &checker.RawResults{
+				ContributorsResults: checker.ContributorsData{
+					Users: []clients.User{
+						{
+							Companies: []string{"comp1", "comp2"},
+							Organizations: []clients.User{
+								{
+									Login:            "Login1",
+									NumContributions: 10,
+								},
+							},
+							NumContributions: 10,
+						},
+						{
+							Companies: []string{"comp3", "comp4"},
+							Organizations: []clients.User{
+								{
+									Login:            "Login2",
+									NumContributions: 10,
+								},
+							},
+							NumContributions: 2,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+				finding.OutcomePositive,
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			findings, s, err := Run(tt.raw)
```

Comment:
let's create a tracking issue to consolidate this validation into a common shared function, since it will be re-used by all probes. Create it and add `// TODO(number): Use common function`

Reply:
Tracked in https://github.com/ossf/scorecard/issues/3472